pymc_marketing/mlflow.py

```diff
@@ -412,14 +478,324 @@ def log_inference_data(
+        extend_idata: bool = False,
+        combined: bool = True,
+        include_last_observations: bool = False,
+        original_scale: bool = True,
+        var_names: list[str] | None = None,
+        **sample_kwargs: dict,
+    ):
+        self.model = model
+        self.predict_method = predict_method
+        self.extend_idata = extend_idata
+        self.combined = combined
+        self.include_last_observations = include_last_observations
+        self.original_scale = original_scale
+        self.var_names = (
+            var_names if var_names is not None else [model.output_var]
+        )  # Initialize if not provided
+        self.sample_kwargs = sample_kwargs
+
+    def predict(
+        self, context: Any, model_input, params: dict[str, Any] | None = None
+    ) -> Any:
+        """Perform predictions or sampling using the specified prediction method.
+
+        Parameters
+        ----------
+        context : Any
+            The context in which the model is running. Isn't specified by users but is passed by MLflow.
+        model_input : array, shape (n_pred, n_features)
+            The input data used for prediction.
+        params : dict, optional
+            A dictionary of parameters to specify the prediction method.
+
+        Returns
+        -------
+        ndarray or InferenceData
+            The predictions or samples generated by the model.
+
+        Raises
+        ------
+        ValueError
+            If an unsupported prediction method is specified.
+
+        """
+        # Use the class-level predict_method if params is not provided or doesn't contain 'predict_method'
+        params = params or {"predict_method": "predict"}
+        predict_method = params.get("predict_method", self.predict_method)
+
+        if predict_method == "predict":
+            return self.model.predict(
+                model_input,
+                extend_idata=self.extend_idata,
+                include_last_observations=self.include_last_observations,
+                original_scale=self.original_scale,
+                var_names=self.var_names,
+                **self.sample_kwargs,
+            )
+        elif predict_method == "sample_posterior_predictive":
+            return self.model.sample_posterior_predictive(
+                model_input,
+                extend_idata=self.extend_idata,
+                combined=self.combined,
+                include_last_observations=self.include_last_observations,
+                original_scale=self.original_scale,
+                var_names=self.var_names,
+                **self.sample_kwargs,
+            )
+        elif predict_method == "sample_prior_predictive":
+            return self.model.sample_prior_predictive(
+                model_input,
+                extend_idata=self.extend_idata,
+                combined=self.combined,
+                var_names=self.var_names,
+                **self.sample_kwargs,  # type: ignore[arg-type]
```

Comment:
Why don't you have this `# type: ignore[arg-type]` in the functions above?

Reply:
Yep good spot, added! 🙇